### ISSUE TITLE
Unable to load DLL 'e_sqlite3': The specified module or one of its dependencies could not be found.

### ISSUE THREAD

**Issue by wk-j** [filipw/dotnet-script#210]
I try to execute this EF code but not success. I guess there is a native DLL was not load correctly into dotnet-script but I have no idea how to fix.

#### Script `Db.csx`

```csharp
#! "netcoreapp2.0"
#r "nuget:NetStandard.Library,2.0.0"
#r "nuget:Microsoft.EntityFrameworkCore,2.0.1"
#r "nuget:Microsoft.EntityFrameworkCore.Sqlite,2.0.1"

using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;

class Order { 
    public int Id { set;get; }
    public List<OrderDetail> Details { set; get; } = new List<OrderDetail>();
}

class OrderDetail { 
    public int Id { set;get; }
    public string Product { set;get; }
}

class OrderContext : DbContext { 
    public DbSet<Order> Orders { set;get; }
    public DbSet<OrderDetail> OrderDetails { set;get; }
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) {
        optionsBuilder.UseSqlite("Data Source=Order.db");
    }
}

var context = new OrderContext();
context.Database.EnsureCreated();
context.Orders.Add(new Order { });
context.SaveChanges();
```

#### Output Message

```
❯ dotnet script "./scripts/Db.csx"
System.TypeInitializationException: The type initializer for 'Microsoft.Data.Sqlite.SqliteConnection' threw an exception. ---> System.Reflection.TargetInvo
cationException: Exception has been thrown by the target of an invocation. ---> System.DllNotFoundException: Unable to load DLL 'e_sqlite3': The specified
module or one of its dependencies could not be found.
 (Exception from HRESULT: 0x8007007E)
   at SQLitePCL.SQLite3Provider_e_sqlite3.NativeMethods.sqlite3_libversion_number()
   at SQLitePCL.SQLite3Provider_e_sqlite3.SQLitePCL.ISQLite3Provider.sqlite3_libversion_number()
   at SQLitePCL.raw.SetProvider(ISQLite3Provider imp)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at Microsoft.Data.Sqlite.Utilities.BundleInitializer.Initialize()
   at Microsoft.Data.Sqlite.SqliteConnection..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqliteRelationalConnection.CreateDbConnection()
   at Microsoft.EntityFrameworkCore.Internal.LazyRef`1.get_Value()
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.Open(Boolean errorsExpected)
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqliteRelationalConnection.Open(Boolean errorsExpected)
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqliteDatabaseCreator.Exists()
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.EnsureCreated()
   at Submission#0.<<Initialize>>d__0.MoveNext() in /Users/wk/Source/DotNetEF/scripts/Db.csx:line 28
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.Scripting.ScriptExecutionState.<RunSubmissionsAsync>d__9`1.MoveNext()
```

**Comment by seesharper** [comment]
Is this on Windows?

**Comment by wk-j** [comment]
On mac OS

```bash
❯ dotnet-script -v

0.17.0

❯ dotnet --info
.NET Command Line Tools (2.1.2)

Product Information:
 Version:            2.1.2
 Commit SHA-1 hash:  5695315371

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.1.2/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df
```

**Comment by seesharper** [comment]
Well, that explains it. We have no story for loading native assets on •nix yet. It should work on windows though

**Comment by seesharper** [comment]
We should document this as a known issue

**Comment by SlowLogicBoy** [comment]
Looks like I duplicated this issue: #318, this should be fixed after #319

**Comment by SlowLogicBoy** [comment]
https://github.com/filipw/dotnet-script/blob/master/src/Dotnet.Script.Core/ScriptCompiler.cs#L28
For linux:
```csharp
[DllImport("libdl.so")]
protected static extern IntPtr dlopen(string filename, int flags);

const int RTLD_NOW = 2; // for dlopen's flags
var handle = dlopen(modulePath, RTLD_NOW);
```

Haven't tested this myself.

**Comment by seesharper** [comment]
Closed by #383
